Gap-detection and slew

amoss opened the pull request:
### Summary

This PR replaces PR9170 completely.
Closes #8921 
Closes #9063 
Fixes #8825

This PR adds (inactive) code to calculate clock-slew and gap-size when slaves reconnect to a master. This code will be activated in a later PR to fill the gaps in the charts.

The streaming component has been heavily restructured to fix several current bugs, and improve the concurrency protection.

* Incorrect handling of return codes
* Uses circular buffer for transmission to avoid starvation on high-latency link
* Remove the warnings from the time defines that were shown for every file on the mac.
* Restructured the sender and receiver, collected their state outside of the host into separate structures that can be locked.
* Made the locking around the sender buffer consistent.
* Fix a case where the stream version was not negotiated correctly.
* Adds bi-directional comms on the streaming and code to receive commands in the sender.
* Cleans up the SSL / raw-socket code and moves it out of the plugins.d parser
* Tracks liveness on the slave connections (seconds since last message received).
* On connection collisions it chooses the new/old socket depending on liveness.

##### Component Name
streaming

##### Test Plan

* Test high- and low- bandwidth settings.
* Used local containers and containers on a remote machine
* Running master on bare-metal

Test A:
1. Connect slaves (observe that no gap is detected on first connection)
2. Pause the containers and kill the socket using `ss -K dst 192.168.1.x`
3. Restart the containers 

Test B:
1. Connect slaves (observe that no gap is detected on first connection)
2. Pull the network cable
3. Reconnect

Test C:
1. Setup multiple containers with the same GUID and mix tests (1) and (2) to check the new collision detection code.
 
##### Additional Information

* We should coordinate about this on Monday, it probably needs to be merged before it blocks any work in the persistent metadata task, but we need to be reasonably sure that the new code is stable.

thiagoftsm commented:
@amoss ,

During more than 8 hours, I observed the stream was relatively stable when we see the the data on dashboard, but I had disconnections when I saw the logs. Considering these disconnections that I observed, these specifics log on `master` called my attention:

```
/var/log/netdata/error.log:2020-05-29 16:06:19: netdata ERROR : WEB_SERVER[static4] : STREAM [receive from [192.168.0.23]:41042]: too busy to accept new streaming request. Will be allowed in 2 secs. (errno 11, Resource temporarily unavailable)
/var/log/netdata/error.log:2020-05-29 16:06:26: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41256] : requested a BEGIN on chart 'A', which does not exist on host 'athenas'. Disabling it.
/var/log/netdata/error.log:2020-05-29 16:06:26: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41256] : STREAM athenas [receive from [192.168.0.23]:41256]: disconnected (completed 51 updates).
/var/log/netdata/error.log:2020-05-29 16:27:09: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41258] : read failed: end of file
/var/log/netdata/error.log:2020-05-29 16:27:09: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41258] : STREAM athenas [receive from [192.168.0.23]:41258]: disconnected (completed 284155 updates).
/var/log/netdata/error.log:2020-05-29 16:27:13: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41268] : Unknown keyword [d.plugin"]
/var/log/netdata/error.log:2020-05-29 16:27:13: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41268] : STREAM athenas [receive from [192.168.0.23]:41268]: disconnected (completed 0 updates).
/var/log/netdata/error.log:2020-05-29 16:30:38: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41270] : read failed: end of file
/var/log/netdata/error.log:2020-05-29 16:30:38: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41270] : STREAM athenas [receive from [192.168.0.23]:41270]: disconnected (completed 20511 updates).
/var/log/netdata/error.log:2020-05-29 16:32:59: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41304] : requested a BEGIN on chart 'sensors.coret', which does not exist on host 'athenas'. Disabling it.
/var/log/netdata/error.log:2020-05-29 16:32:59: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41304] : STREAM athenas [receive from [192.168.0.23]:41304]: disconnected (completed 21887 updates).
/var/log/netdata/error.log:2020-05-29 16:33:00: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41310] : Unknown keyword [ = 0
/var/log/netdata/error.log:2020-05-29 16:33:00: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41310] : STREAM athenas [receive from [192.168.0.23]:41310]: disconnected (completed 0 updates).
/var/log/netdata/error.log:2020-05-29 16:36:46: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41312] : requested a BEGIN on chart 'ebp', which does not exist on host 'athenas'. Disabling it.
/var/log/netdata/error.log:2020-05-29 16:36:46: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41312] : STREAM athenas [receive from [192.168.0.23]:41312]: disconnected (completed 20370 updates).
/var/log/netdata/error.log:2020-05-29 16:36:49: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41354] : requested an END, without a BEGIN on host 'athenas'. Disabling it.
/var/log/netdata/error.log:2020-05-29 16:36:49: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41354] : STREAM athenas [receive from [192.168.0.23]:41354]: disconnected (completed 0 updates).
/var/log/netdata/error.log:2020-05-29 16:39:08: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41356] : read failed: end of file
/var/log/netdata/error.log:2020-05-29 16:39:08: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41356] : STREAM athenas [receive from [192.168.0.23]:41356]: disconnected (completed 19080 updates).
/var/log/netdata/error.log:2020-05-29 16:39:14: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41368] : Unknown keyword [ds"]
/var/log/netdata/error.log:2020-05-29 16:39:14: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41368] : STREAM athenas [receive from [192.168.0.23]:41368]: disconnected (completed 0 updates).
/var/log/netdata/error.log:2020-05-29 16:39:15: netdata ERROR : WEB_SERVER[static4] : STREAM [receive from [192.168.0.23]:41370]: too busy to accept new streaming request. Will be allowed in 1 secs. (errno 11, Resource temporarily unavailable)
/var/log/netdata/error.log:2020-05-29 16:41:00: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41372] : requested a SET on chart 'netdata.plugin_proc_modules' of host 'athenas', without a dimension. Disabling it.
/var/log/netdata/error.log:2020-05-29 16:41:00: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41372] : STREAM athenas [receive from [192.168.0.23]:41372]: disconnected (completed 9304 updates).
/var/log/netdata/error.log:2020-05-29 16:41:05: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41384] : Unknown keyword [eq"]
/var/log/netdata/error.log:2020-05-29 16:41:05: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41384] : STREAM athenas [receive from [192.168.0.23]:41384]: disconnected (completed 0 updates).
/var/log/netdata/error.log:2020-05-29 16:41:05: netdata ERROR : WEB_SERVER[static4] : STREAM [receive from [192.168.0.23]:41386]: too busy to accept new streaming request. Will be allowed in 2 secs. (errno 11, Resource temporarily unavailable)
/var/log/netdata/error.log:2020-05-29 16:43:12: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41388] : read failed: end of file
/var/log/netdata/error.log:2020-05-29 16:43:12: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41388] : STREAM athenas [receive from [192.168.0.23]:41388]: disconnected (completed 11077 updates).
/var/log/netdata/error.log:2020-05-29 16:43:14: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41404] : Unknown keyword [emp-isa-0000_temp3"]
/var/log/netdata/error.log:2020-05-29 16:43:14: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41404] : STREAM athenas [receive from [192.168.0.23]:41404]: disconnected (completed 0 updates).
/var/log/netdata/error.log:2020-05-29 16:45:21: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41406] : read failed: end of file
/var/log/netdata/error.log:2020-05-29 16:45:21: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41406] : STREAM athenas [receive from [192.168.0.23]:41406]: disconnected (completed 9021 updates).
/var/log/netdata/error.log:2020-05-29 16:45:25: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41430] : Unknown keyword [st"]
/var/log/netdata/error.log:2020-05-29 16:45:25: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41430] : STREAM athenas [receive from [192.168.0.23]:41430]: disconnected (completed 0 updates).
/var/log/netdata/error.log:2020-05-29 16:45:26: netdata ERROR : WEB_SERVER[static3] : STREAM [receive from [192.168.0.23]:41432]: too busy to accept new streaming request. Will be allowed in 1 secs. (errno 11, Resource temporarily unavailable)
/var/log/netdata/error.log:2020-05-29 16:47:36: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41434] : read failed: end of file
/var/log/netdata/error.log:2020-05-29 16:47:36: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41434] : STREAM athenas [receive from [192.168.0.23]:41434]: disconnected (completed 17930 updates).
/var/log/netdata/error.log:2020-05-29 16:47:37: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41442] : requested a SET on dimension kernel with value 0 on host 'athenas', without a BEGIN. Disabling it.
/var/log/netdata/error.log:2020-05-29 16:47:37: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41442] : STREAM athenas [receive from [192.168.0.23]:41442]: disconnected (completed 0 updates).
/var/log/netdata/error.log:2020-05-29 16:47:37: netdata ERROR : WEB_SERVER[static3] : STREAM [receive from [192.168.0.23]:41444]: too busy to accept new streaming request. Will be allowed in 2 secs. (errno 11, Resource temporarily unavailable)
/var/log/netdata/error.log:2020-05-29 16:49:13: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41446] : requested a SET on chart 'system.softirqs' of host 'athenas', without a dimension. Disabling it.
/var/log/netdata/error.log:2020-05-29 16:49:13: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41446] : STREAM athenas [receive from [192.168.0.23]:41446]: disconnected (completed 9387 updates).
/var/log/netdata/error.log:2020-05-29 16:49:15: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41452] : Unknown keyword [sed"]
/var/log/netdata/error.log:2020-05-29 16:49:15: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41452] : STREAM athenas [receive from [192.168.0.23]:41452]: disconnected (completed 0 updates).
/var/log/netdata/error.log:2020-05-29 16:50:52: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41454] : read failed: end of file
/var/log/netdata/error.log:2020-05-29 16:50:52: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41454] : STREAM athenas [receive from [192.168.0.23]:41454]: disconnected (completed 8069 updates).
/var/log/netdata/error.log:2020-05-29 16:52:58: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41466] : read failed: end of file
/var/log/netdata/error.log:2020-05-29 16:52:58: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41466] : STREAM athenas [receive from [192.168.0.23]:41466]: disconnected (completed 3364 updates).
/var/log/netdata/error.log:2020-05-29 16:52:58: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41496] : Unknown keyword [reads"]
/var/log/netdata/error.log:2020-05-29 16:52:58: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41496] : STREAM athenas [receive from [192.168.0.23]:41496]: disconnected (completed 0 updates).
/var/log/netdata/error.log:2020-05-29 16:52:59: netdata ERROR : WEB_SERVER[static1] : STREAM [receive from [192.168.0.23]:41498]: too busy to accept new streaming request. Will be allowed in 1 secs. (errno 11, Resource temporarily unavailable)
/var/log/netdata/error.log:2020-05-29 16:54:51: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41500] : read failed: end of file
/var/log/netdata/error.log:2020-05-29 16:54:51: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41500] : STREAM athenas [receive from [192.168.0.23]:41500]: disconnected (completed 14751 updates).
/var/log/netdata/error.log:2020-05-29 16:54:51: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41506] : Unknown keyword [ES"]
/var/log/netdata/error.log:2020-05-29 16:54:51: netdata ERROR : STREAM_RECEIVER[athenas,[192.168.0.23]:41506] : STREAM athenas [receive from [192.168.0.23]:41506]: disconnected (completed 0 updates).
```

This was the worse period of time for the stream, the rest was stable with few disconnections probably related to hardware problems. 

On the slave side we had this:

```
/var/log/netdata/error.log:2020-05-29 16:05:23: netdata ERROR : STATSD : STREAM athenas [send]: not ready - discarding collected metrics. (errno 22, Invalid argument)
/var/log/netdata/error.log:2020-05-29 16:05:28: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:05:35: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: error during read (-1). Restarting connection (errno 104, Connection reset by peer)
/var/log/netdata/error.log:2020-05-29 16:25:58: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: buffer full (1048576-bytes) after 36724253 bytes. Restarting connection
/var/log/netdata/error.log:2020-05-29 16:25:58: netdata ERROR : WEB_SERVER[static1] : STREAM athenas [send]: not ready - discarding collected metrics. (errno 2, No such file or directory)
/var/log/netdata/error.log:2020-05-29 16:25:58: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:26:03: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:26:08: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:26:17: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:26:22: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 575585 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:26:22: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: failed to send metrics - closing connection - we have sent 408336 bytes on this connection. (errno 32, Broken pipe)
/var/log/netdata/error.log:2020-05-29 16:26:22: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: restart stream because socket reports errors (POLLERR) - 408336 bytes transmitted.
/var/log/netdata/error.log:2020-05-29 16:26:24: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 65133 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:28:24: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: buffer full (1048576-bytes) after 2672259 bytes. Restarting connection
/var/log/netdata/error.log:2020-05-29 16:28:24: netdata ERROR : WEB_SERVER[static1] : STREAM athenas [send]: not ready - discarding collected metrics.
/var/log/netdata/error.log:2020-05-29 16:28:24: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:28:30: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:28:36: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:28:41: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:28:47: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:28:52: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:28:58: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:29:03: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:29:08: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:29:13: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:29:18: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:29:23: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:29:30: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:29:35: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:29:40: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:29:46: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:29:51: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 1048572 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:31:57: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: buffer full (1048576-bytes) after 2849668 bytes. Restarting connection
/var/log/netdata/error.log:2020-05-29 16:31:57: netdata ERROR : PLUGINSD[python.d] : STREAM athenas [send]: not ready - discarding collected metrics. (errno 9, Bad file descriptor)
/var/log/netdata/error.log:2020-05-29 16:31:58: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:32:04: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:32:09: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 296064 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:32:09: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: failed to send metrics - closing connection - we have sent 414128 bytes on this connection. (errno 32, Broken pipe)
/var/log/netdata/error.log:2020-05-29 16:32:09: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: restart stream because socket reports errors (POLLERR) - 414128 bytes transmitted.
/var/log/netdata/error.log:2020-05-29 16:32:09: netdata ERROR : WEB_SERVER[static1] : STREAM athenas [send]: not ready - discarding collected metrics.
/var/log/netdata/error.log:2020-05-29 16:32:11: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 465300 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:34:10: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: buffer full (1048576-bytes) after 2654498 bytes. Restarting connection
/var/log/netdata/error.log:2020-05-29 16:34:10: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:34:10: netdata ERROR : WEB_SERVER[static1] : STREAM athenas [send]: not ready - discarding collected metrics.
/var/log/netdata/error.log:2020-05-29 16:34:15: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:34:21: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:34:26: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:34:31: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:34:36: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:34:41: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:34:46: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:34:51: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:34:56: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:35:01: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:35:06: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:35:11: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:35:18: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:35:23: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:35:28: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:35:37: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:35:42: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:35:47: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:35:53: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:35:58: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 660378 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:36:00: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 508 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:37:54: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: buffer full (1048576-bytes) after 2487798 bytes. Restarting connection
/var/log/netdata/error.log:2020-05-29 16:37:54: netdata ERROR : WEB_SERVER[static1] : STREAM athenas [send]: not ready - discarding collected metrics.
/var/log/netdata/error.log:2020-05-29 16:37:55: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:38:00: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:38:05: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:38:11: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:38:16: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:38:23: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 269239 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:38:23: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: failed to send metrics - closing connection - we have sent 438744 bytes on this connection. (errno 32, Broken pipe)
/var/log/netdata/error.log:2020-05-29 16:38:23: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: restart stream because socket reports errors (POLLERR) - 438744 bytes transmitted.
/var/log/netdata/error.log:2020-05-29 16:38:24: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:38:32: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 341048 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:39:46: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: buffer full (1048576-bytes) after 1254092 bytes. Restarting connection
/var/log/netdata/error.log:2020-05-29 16:39:46: netdata ERROR : WEB_SERVER[static1] : STREAM athenas [send]: not ready - discarding collected metrics.
/var/log/netdata/error.log:2020-05-29 16:39:47: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:39:52: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:39:57: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:40:02: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:40:08: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:40:13: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 63268 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:40:13: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: failed to send metrics - closing connection - we have sent 408336 bytes on this connection. (errno 32, Broken pipe)
/var/log/netdata/error.log:2020-05-29 16:40:13: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: restart stream because socket reports errors (POLLERR) - 408336 bytes transmitted.
/var/log/netdata/error.log:2020-05-29 16:40:14: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:40:19: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 577466 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:41:40: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: buffer full (1048576-bytes) after 1475196 bytes. Restarting connection
/var/log/netdata/error.log:2020-05-29 16:41:40: netdata ERROR : WEB_SERVER[static1] : STREAM athenas [send]: not ready - discarding collected metrics.
/var/log/netdata/error.log:2020-05-29 16:41:40: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:41:46: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:41:51: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:41:59: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:42:07: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:42:12: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:42:17: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:42:23: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 684730 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:42:23: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: failed to send metrics - closing connection - we have sent 363821 bytes on this connection. (errno 32, Broken pipe)
/var/log/netdata/error.log:2020-05-29 16:42:23: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: restart stream because socket reports errors (POLLERR) - 363821 bytes transmitted.
/var/log/netdata/error.log:2020-05-29 16:42:25: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 508 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:43:38: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: buffer full (1048576-bytes) after 1219299 bytes. Restarting connection
/var/log/netdata/error.log:2020-05-29 16:43:38: netdata ERROR : WEB_SERVER[static1] : STREAM athenas [send]: not ready - discarding collected metrics.
/var/log/netdata/error.log:2020-05-29 16:43:38: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:43:43: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:43:48: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:43:54: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:43:59: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:44:04: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:44:09: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:44:14: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:44:19: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:44:24: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:44:29: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:44:34: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 513524 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:44:34: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: restart stream because socket reports errors (POLLERR) - 411232 bytes transmitted.
/var/log/netdata/error.log:2020-05-29 16:44:35: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:44:41: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 124221 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:46:30: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: buffer full (1048576-bytes) after 2341844 bytes. Restarting connection
/var/log/netdata/error.log:2020-05-29 16:46:30: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:46:30: netdata ERROR : WEB_SERVER[static1] : STREAM athenas [send]: not ready - discarding collected metrics.
/var/log/netdata/error.log:2020-05-29 16:46:35: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:46:40: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:46:45: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 268431 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:46:46: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:46:51: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 366482 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:48:05: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: buffer full (1048576-bytes) after 1263257 bytes. Restarting connection
/var/log/netdata/error.log:2020-05-29 16:48:05: netdata ERROR : WEB_SERVER[static1] : STREAM athenas [send]: not ready - discarding collected metrics.
/var/log/netdata/error.log:2020-05-29 16:48:05: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:48:14: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:48:24: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 53285 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:48:24: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: error during read (-1). Restarting connection (errno 104, Connection reset by peer)
/var/log/netdata/error.log:2020-05-29 16:48:24: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: failed to send metrics - closing connection - we have sent 438744 bytes on this connection. (errno 9, Bad file descriptor)
/var/log/netdata/error.log:2020-05-29 16:48:24: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: restart stream because socket reports errors (POLLERR) - 438744 bytes transmitted.
/var/log/netdata/error.log:2020-05-29 16:48:25: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 557032 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:49:34: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: buffer full (1048576-bytes) after 1101376 bytes. Restarting connection
/var/log/netdata/error.log:2020-05-29 16:49:34: netdata ERROR : WEB_SERVER[static3] : STREAM athenas [send]: not ready - discarding collected metrics.
/var/log/netdata/error.log:2020-05-29 16:49:34: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:49:39: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:49:44: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:49:49: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:49:57: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:50:03: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 1048491 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:50:53: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: buffer full (1048576-bytes) after 517022 bytes. Restarting connection
/var/log/netdata/error.log:2020-05-29 16:50:53: netdata ERROR : PLUGINSD[apps] : STREAM athenas [send]: not ready - discarding collected metrics. (errno 9, Bad file descriptor)
/var/log/netdata/error.log:2020-05-29 16:50:53: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:50:58: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:51:03: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:51:08: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:51:13: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:51:19: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:51:24: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:51:29: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:51:35: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:51:40: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:51:45: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:51:50: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:51:55: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:52:01: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:52:07: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 531639 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:52:07: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: failed to send metrics - closing connection - we have sent 438744 bytes on this connection. (errno 32, Broken pipe)
/var/log/netdata/error.log:2020-05-29 16:52:08: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:52:13: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 78659 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:53:49: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: buffer full (1048576-bytes) after 1930526 bytes. Restarting connection
/var/log/netdata/error.log:2020-05-29 16:53:49: netdata ERROR : PLUGIN[proc] : STREAM athenas [send]: not ready - discarding collected metrics. (errno 2, No such file or directory)
/var/log/netdata/error.log:2020-05-29 16:53:50: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:53:55: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:54:00: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 697799 bytes of metrics already in the buffer.
/var/log/netdata/error.log:2020-05-29 16:54:00: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: failed to send metrics - closing connection - we have sent 350718 bytes on this connection. (errno 32, Broken pipe)
/var/log/netdata/error.log:2020-05-29 16:54:00: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: restart stream because socket reports errors (POLLERR) - 350718 bytes transmitted.
/var/log/netdata/error.log:2020-05-29 16:54:00: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send to 192.168.0.17:20000]: server is not replying properly (is it a netdata?).
/var/log/netdata/error.log:2020-05-29 16:54:06: netdata ERROR : STREAM_SENDER[athenas] : STREAM athenas [send]: discarding 508 bytes of metrics already in the buffer.

```

thiagoftsm commented:
When I compiled this branch I received these warnings:

```
streaming/receiver.c: In function 'streaming_timestamp':
streaming/receiver.c:70:13: warning: variable 'ret' set but not used [-Wunused-but-set-variable]
   70 |         int ret;
      |             ^~~
streaming/receiver.c: In function 'rrdpush_receive':
streaming/receiver.c:223:15: warning: assignment discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  223 |     rpt->tags = appconfig_set_default(&stream_config, rpt->machine_guid, "host tags", (rpt->tags)?rpt->tags:"");
``` 

I compiled using the following C flags:

```
CFLAGS="-O1 -ggdb -Wall -Wextra -Wformat-signedness -fstack-protector-all -DNETDATA_INTERNAL_CHECKS=1 -D_FORTIFY_SOURCE=2 -DNETDATA_VERIFY_LOCKS=1" 
```

thiagoftsm commented:
@amoss yesterday I tested this branch that has the code from the previous and some additional codes and it looks like that the new codes that you brought here ( I did not review them yet) could solve the problems I reported in my first comment on this PR, because I did not have a large problem with connections. This time I had the exactly number of disconnections I always had with the previous streams for the same length of time.

stelfrag commented on streaming/receiver.c:
```diff
@@ -0,0 +1,417 @@
+    while(!netdata_exit);
+done:
+    info("PARSER ended");
+    return ((PARSER_USER_OBJECT *) user)->count;
```
This will leak memory (user and parser) objects

amoss replied:
Fixed

stelfrag commented:
- Started master with the updated code
- Started slave (old version)
- Paused master (suspend)
- Resumed master
  - Slave attempts to connect but keeps failing with 

```
2020-06-01 17:38:42: netdata INFO  : WEB_SERVER[static5] : clients wants to STREAM metrics.
2020-06-01 17:38:42: netdata INFO  : STREAM_RECEIVER[homeroot,[192.168.1.5]:59752] : thread created with task id 116188
2020-06-01 17:38:42: netdata INFO  : STREAM_RECEIVER[homeroot,[192.168.1.5]:59752] : set name of thread 116188 to STREAM_RECEIVER
2020-06-01 17:38:42: netdata INFO  : STREAM_RECEIVER[homeroot,[192.168.1.5]:59752] : STREAM homeroot [192.168.1.5]:59752: receive thread created (task id 116188)
2020-06-01 17:38:42: netdata INFO  : STREAM_RECEIVER[homeroot,[192.168.1.5]:59752] : STREAM homeroot [receive from [192.168.1.5]:59752]: client willing to stream metrics for host 'homeroot' with machine_guid 'e574ee06-6e83-11ea-98ac-b827eb6f70a8': update every = 1, history = 3996, memory mode = dbengine, health enabled, tags ''
2020-06-01 17:38:42: netdata INFO  : STREAM_RECEIVER[homeroot,[192.168.1.5]:59752] : STREAM homeroot [receive from [192.168.1.5]:59752]: initializing communication...
2020-06-01 17:38:42: netdata INFO  : STREAM_RECEIVER[homeroot,[192.168.1.5]:59752] : STREAM homeroot [receive from [192.168.1.5]:59752]: Netdata is using the stream version 2.
2020-06-01 17:38:42: netdata INFO  : STREAM_RECEIVER[homeroot,[192.168.1.5]:59752] : STREAM homeroot [receive from [192.168.1.5]:59752]: multiple streaming connections for the same host detected. Rejecting new connection.
2020-06-01 17:38:42: netdata INFO  : STREAM_RECEIVER[homeroot,[192.168.1.5]:59752] : STREAM homeroot [receive from [192.168.1.5]:59752]: receive thread ended (task id 116188)
2020-06-01 17:38:42: netdata INFO  : STREAM_RECEIVER[homeroot,[192.168.1.5]:59752] : thread with task id 116188 finished
2020-06-01 17:38:43: netdata INFO  : WEB_SERVER[static2] : clients wants to STREAM metrics.
2020-06-01 17:38:43: netdata INFO  : STREAM_RECEIVER[homeroot,[192.168.1.5]:59754] : thread created with task id 116189
2020-06-01 17:38:43: netdata INFO  : STREAM_RECEIVER[homeroot,[192.168.1.5]:59754] : set name of thread 116189 to STREAM_RECEIVER
2020-06-01 17:38:43: netdata INFO  : STREAM_RECEIVER[homeroot,[192.168.1.5]:59754] : STREAM homeroot [192.168.1.5]:59754: receive thread created (task id 116189)
2020-06-01 17:38:43: netdata INFO  : STREAM_RECEIVER[homeroot,[192.168.1.5]:59754] : STREAM homeroot [receive from [192.168.1.5]:59754]: client willing to stream metrics for host 'homeroot' with machine_guid 'e574ee06-6e83-11ea-98ac-b827eb6f70a8': update every = 1, history = 3996, memory mode = dbengine, health enabled, tags ''
2020-06-01 17:38:43: netdata INFO  : STREAM_RECEIVER[homeroot,[192.168.1.5]:59754] : STREAM homeroot [receive from [192.168.1.5]:59754]: initializing communication...
2020-06-01 17:38:43: netdata INFO  : STREAM_RECEIVER[homeroot,[192.168.1.5]:59754] : STREAM homeroot [receive from [192.168.1.5]:59754]: Netdata is using the stream version 2.
2020-06-01 17:38:43: netdata INFO  : STREAM_RECEIVER[homeroot,[192.168.1.5]:59754] : STREAM homeroot [receive from [192.168.1.5]:59754]: multiple streaming connections for the same host detected. Rejecting new connection.
2020-06-01 17:38:43: netdata INFO  : STREAM_RECEIVER[homeroot,[192.168.1.5]:59754] : STREAM homeroot [receive from [192.168.1.5]:59754]: receive thread ended (task id 116189)
2020-06-01 17:38:43: netdata INFO  : STREAM_RECEIVER[homeroot,[192.168.1.5]:59754] : thread with task id 116189 finished
```

stelfrag commented:
I broke the compilation process (not wrapped with ifdef HTTPS) with the parser when using `--disable-https`  so the following need to go away from `plugins_d.h`  (the functions do not exist anymore) 

```
extern int pluginsd_update_buffer(char *output, SSL *ssl);
extern char * pluginsd_get_from_buffer(char *output, int *bytesread, char *input, SSL *ssl, char *src);
```

amoss commented:
I'm redoing some testing to see if I could recreate the problems that @thiagoftsm saw with truncated keywords on a laggy connection. I found this, which does not relate to the gaps that we are trying to fill but seems to be an architectural issue;

![image](https://user-images.githubusercontent.com/1043609/83436173-ad8ed200-a43d-11ea-922e-62ed77a49f5b.png)

The test-case is to move a laptop to the edge of a wifi zone to get a network connection with limited bandwidth and high latency (from the message re-transmission). What happens is that the slave cannot transmit data fast enough to fill the chart, the master doesn't know which samples it can see and which are missing so the gaps on the charts correspond to the dead-space on the connection and the filled in parts are the times when fragments were received over the connection.

amoss commented:
Also, I've recreated the problem that @thiagoftsm saw - it seems that the last message before the link drops out is trashed, e.g.:
```
2020-06-01 19:22:51: netdata ERROR : STREAM_RECEIVER[eca941772753,[192.168.1.223]:55861] : (0291@parser/par:parser_action  ): Unknown keyword [1]
2020-06-01 19:22:51: netdata INFO  : STREAM_RECEIVER[eca941772753,[192.168.1.223]:55861] : (0179@streaming/:streaming_parse): PARSER ended
2020-06-01 19:22:51: netdata ERROR : STREAM_RECEIVER[eca941772753,[192.168.1.223]:55861] : (0386@streaming/:rrdpush_receive): STREAM eca941772753 [receive from [192.168.1.223]:55861]: disconnected (completed 0 updates).
2020-06-01 19:22:51: netdata INFO  : STREAM_RECEIVER[eca941772753,[192.168.1.223]:55861] : (0019@streaming/:rrdpush_receive): STREAM eca941772753 [receive from [192.168.1.223]:55861]: receive thread ended (task id 7359)
```

amoss commented:
@stelfrag I've seen the same problem with reconnections - there was another check that rejected the salve later in the process that I missed. Will upload a patch soon.

amoss commented:
> I broke the compilation process (not wrapped with ifdef HTTPS) with the parser when using `--disable-https` so the following need to go away from `plugins_d.h` (the functions do not exist anymore)
> 
> ```
> extern int pluginsd_update_buffer(char *output, SSL *ssl);
> extern char * pluginsd_get_from_buffer(char *output, int *bytesread, char *input, SSL *ssl, char *src);
> ```

Oops. Missed those ones.... Will kill with fire...

thiagoftsm commented:
> Also, I've recreated the problem that @thiagoftsm saw - it seems that the last message before the link drops out is trashed, e.g.:
> 
> ```
> 2020-06-01 19:22:51: netdata ERROR : STREAM_RECEIVER[eca941772753,[192.168.1.223]:55861] : (0291@parser/par:parser_action  ): Unknown keyword [1]
> 2020-06-01 19:22:51: netdata INFO  : STREAM_RECEIVER[eca941772753,[192.168.1.223]:55861] : (0179@streaming/:streaming_parse): PARSER ended
> 2020-06-01 19:22:51: netdata ERROR : STREAM_RECEIVER[eca941772753,[192.168.1.223]:55861] : (0386@streaming/:rrdpush_receive): STREAM eca941772753 [receive from [192.168.1.223]:55861]: disconnected (completed 0 updates).
> 2020-06-01 19:22:51: netdata INFO  : STREAM_RECEIVER[eca941772753,[192.168.1.223]:55861] : (0019@streaming/:rrdpush_receive): STREAM eca941772753 [receive from [192.168.1.223]:55861]: receive thread ended (task id 7359)
> ```

Without any doubt, you could recreate the exact problems I created with my old hardwares.

amoss commented:
Congratulations @stelfrag :) I changed your unknown keyword `error()` and added an `info()` to dump in the `*input` on entry and it seems to be a bug in the parser:
```
2020-06-01 19:59:11: netdata INFO  : STREAM_RECEIVER[b4ba38cf778e,[172.19.0.2]:51778] : (0254@parser/par:parser_action  ): Initial input: SET "HI" = 41
2020-06-01 19:59:11: netdata ERROR : STREAM_RECEIVER[1701d8fc56ee,[192.168.1.223]:56706] : (0293@parser/par:parser_action  ): Unknown keyword [GIN]
```

amoss commented:
Ah I spoke too soon :) I missed the difference in receivers producing the output. I took a slice by a single receiver and it looks like I am passing the wrong input into the parser...

thiagoftsm commented:
@amoss we continue having warnings from the old PR:

```
 --- Compile netdata --- 
[/home/thiago/Netdata/netdata]# make -j4 
streaming/receiver.c: In function 'streaming_timestamp':
streaming/receiver.c:70:13: warning: variable 'ret' set but not used [-Wunused-but-set-variable]
   70 |         int ret;
      |             ^~~
streaming/receiver.c: In function 'rrdpush_receive':
streaming/receiver.c:228:15: warning: assignment discards 'const' qualifier from pointer target type [-Wdiscarded-qualifiers]
  228 |     rpt->tags = appconfig_set_default(&stream_config, rpt->machine_guid, "host tags", (rpt->tags)?rpt->tags:"");
      |               ^
 OK   
```

thiagoftsm commented:
After the last commit, I  observed that I am having less messages of disconnections and this gave me an impression that the PR became more stable, on the other hand, when I open my dashboard  I am seeing this:

![err69](https://user-images.githubusercontent.com/49162938/83464158-8648fc80-a45f-11ea-803e-e719b14b5b05.png)

I do not have any errors, but the gaps are a constant.

thiagoftsm commented:
I realized another test using TLS now during 3 hours and I verified that the number of disconnections increased 10 times.
The stream is more stable with plain text.

stelfrag commented on streaming/receiver.c:
```diff
@@ -0,0 +1,425 @@
+    //size_t count = pluginsd_process(host, &cd, fp, 1);
+
+    log_stream_connection(rpt->client_ip, rpt->client_port, rpt->key, rpt->host->machine_guid, rpt->host->hostname, "DISCONNECTED");
+    error("STREAM %s [receive from [%s]:%s]: disconnected (completed %zu updates).", rpt->host->hostname, rpt->client_ip, rpt->client_port, count);
```
The fix now now timeouts the old connection in 30 seconds and accepts the new one. But the old one doesn't go away. It is marked as being shutdown, but the thread is there, stuck in fgets

```
#1  __GI___libc_read (fd=213, buf=0x561448eee0d0, nbytes=4096) at ../sysdeps/unix/sysv/linux/read.c:24
24	in ../sysdeps/unix/sysv/linux/read.c

 :
 :
#8  0x0000561444e85478 in streaming_parser (rpt=0x561448ca5060, cd=0x7fe5692c7810, fp=0x561448eec1a0) at streaming/receiver.c:170
170	        if (receiver_read(rpt, fp))
(gdb) p rpt->shutdown 
$1 = 1
```

If for whatever reason the fgets returns, it will immediately disconnect but it shouldn't execute the code in L393-L401 (maybe the `rpt->host->receiver  == rpt` test is enough to protect that part)

mfundul replied:
This is what can actually crash the agent after the socket times out. Maybe there is a way to force the socket to die somehow. Maybe try [shutdown](https://www.man7.org/linux/man-pages/man2/shutdown.2.html)?

amoss replied:
When the `fgets` retiurns it will not touch the host structure, only the `receiver_state` which has been detached from the host. I've added the `shudown()` call to help along towards its destination.

amoss commented:
> I realized another test using TLS now during 3 hours and I verified that the number of disconnections increased 10 times.
> The stream is more stable with plain text.

I need more detail to understand what you are reporting:
*  How many disconnections are you seeing for plain text, how many for TLS? 
* What kind of configuration do you have in terms of masters and slaves? 
* What are the network conditions like?
